install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -1,5 +1,41 @@
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
+metadata:
+  name: machine-config-controller
+  namespace: openshift-machine-config-operator
+  labels:
+    k8s-app: machine-config-controller
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  groups:
+    - name: mcc-paused-pool-kubelet-ca
+      rules:
+        - alert: MachineConfigControllerPausedPoolKubeletCA
+          expr: |
+             sum by (pool) (last_over_time(machine_config_controller_paused_pool_kubelet_ca[5m])) > 0
+          for: 60m
+          labels:
+            severity: warning
+          annotations:
+            summary: "Paused machine configuration pool '{{$labels.pool}}' is blocking a necessary certificate rotation. Unpause this pool before the current kube-api-to-kubelet-signer expires in {{ $value | humanizeDuration }}."
+            description: "Machine config pools have a 'pause' feature, which allows configuration to be rendered, but prevents it from being rolled out to the nodes. Unforunately, this also stops necessary cluster configuration ( like certificate rotations ) from taking place. This alert indicates that a certificate rotation has taken place, and the new kubelect-ca certificate bundle has been rendered into a machine config, but because the pool '{{$labels.pool}} is paused, that machine config cannot be rolled out to the nodes in that pool. As a result, some interactive commands `oc log, oc exec, oc logs, oc attach` will not function against the nodes in the '{{$labels.pool}}'', and the nodes in this pool will stop functioning properly if you do not unpause the pool before the current kube-api-to-kubelet signer expires in {{ $value | humanizeDuration }}."
+            runbook_url: https://github.com/openshift/blob/master/alerts/machine-config-operator/MachineConfigControllerPausedPoolKubeletCA.md
+        - alert: MachineConfigControllerPausedPoolKubeletCA
+          expr: |
+             sum by (pool) (last_over_time(machine_config_controller_paused_pool_kubelet_ca[5m]) - time()) < (86400 * 365) AND sum by (pool) (last_over_time(machine_config_controller_paused_pool_kubelet_ca[5m])) > 0
```

Comment:
**This is set to 365 for testing so it always alerts**. 

For real I was thinking 14 or 21 days but I would love suggestions. I want to give enough time that an outage can be scheduled and completed. 

The thought here was that: 
-  max_over_time ignores zeroes, so we would take 5 minutes to shut off once the condition is fixed
-  last_over_time ignores missing data like max_over_time (so restarts, etc) but pays attention when we get a zero 

So the joint condition lets us avoid flapping while turning off the alert immediately once the pool syncs and sets it back to zero.

Reply:
No strong opinion on exact timing. But yeah, giving 2 weeks of buffer sounds good.

Reply:
should also preserve `namespace` label here as per @simonpasquier previous comment ?  Else LGTM

Reply:
@dofinn From your SRE experience, any thoughts on how many days in advance alert should be moved to critical before cluster gets unresponsive?

Reply:
@sinnykumari I have not encountered this situation yet, one consideration in managed OpenShift space would be hibernation and potential implications of paused pools with respect to this alert. https://issues.redhat.com/browse/SDE-1173

Reply:
That's interesting usecase and will be interesting to tackle the cert rotation problem when cluster is in hibernation for long time